libbeat/outputs/elasticsearch/output.go

```diff
@@ -117,6 +120,30 @@ func (out *elasticsearchOutput) init(
 		return err
 	}
 
+	// Check if template should be loaded
+	// Not being able to load the template will output an error but will not stop execution
+	if config.Template.Name != "" && len(clients) > 0 {
+
+		esClient := clients[0].(*Client)
+
+		logp.Debug("test", "Test: %v", config.Template)
+
+		// Check if template already exist or should be overwritten
+		if !esClient.CheckTemplate(config.Template.Name) && config.Template.Overwrite {
```

Comment:
I also think there is a potential conflict. For different versions the solution is probably to version the templates and indices. I recently updated the issue and add a note about the version problem: https://github.com/elastic/beats/pull/639 The initial solution would be to not make it automatic.